### ISSUE TITLE
Seq.range is causing out of memory

### ISSUE THREAD

**Issue by lijunle** [fsprojects/Fable#105]
### Description

This is a valid code `['1'..'9']` to generate a char list. However, it the Fable-Core's [Seq.range](https://github.com/fsprojects/Fable/blob/master/import/core/fable-core.js#L1236) function does not handle it correctly. Seq.range only handle the case the `first` and `last` are numbers.

### Expected behavior

Expect the code to generate a char list from '1' to '9'.

### Actual behavior

The build is passing, the webpack works fine. Finally, IE gives me out-of-memory error, Chrome hangs for seconds and finally be killed - because the steps continuously going up but never touch the `last` element - it is a char (a string actually after convert to JavaScript).

### Known workarounds

For 1 to 9 digit char, it is fine to write

```fs
[1..9] |> List.map char // CAUTE: this ONLY works in compiled JavaScript, not in F#
```

However, for a to z character, and for code works both in F# and JavaScript, I am writing a full list: 

```fs
let charsA2Z = ['a'; 'b'; 'c'; 'd'; 'e'; 'f'; 'g'; 'h'; 'i'; 'j'; 'k'; 'l'; 'm'; 'n'; 'o'; 'p'; 'q'; 'r'; 's'; 't'; 'u'; 'v'; 'w'; 'x'; 'y'; 'z']
```

Related: https://github.com/fsprojects/Fable/issues/106

**Comment by fdcastel** [comment]
@lijunle Please check if #107 solves it for you.

Note that it requires a change into fable-core.js.

**Comment by lijunle** [comment]
@fdcastel thanks for handling this. I find that not only integer and char can be seq. All type class implementing `IEnumable<T>` can be pass into seq.

So, it could be better to throw an exception for the unsupported type and maybe support then later.

**Comment by fdcastel** [comment]
Yes. Any IEnumerable can be used as a sequence in F#. But I thought you were having problems only with the range constructors. The fix was essentially to solve that.

Could you share what are you trying to accomplish? Maybe we can find a better way :)

**Comment by lijunle** [comment]
Yes, I'm only using char here. But I spent a lot of time to figure out which piece of code of causing the problem. In case someone use another enumerable type in seq, he will spend another couples of hours to find out which piece of code is causing problem. That is why the error message for unsupported type is important and useful.

**Comment by alfonsogarciacaro** [comment]
I completely agree that the error messages should be as informative as possible and we're trying to do that. Unfortunately there're still some blind spots which are difficult to preview in advance. @fdcastel's commit now allows to create ranges from chars (thanks for that!) and it's published in versions fable-compiler 0.2.3 & fable-core 0.0.12 (remember to update both). Please give it a try and feel free to close the issue if it's solved.

By other enumerables I guess you mean it's possible to generate ranges from custom types other than numeric and char values. In theory it should be possible but I couldn't find the exact requirements for that and I'm not sure there'll be much practical value for that. If this is not exactly what you meant, could you please provide a code example and the error message you would expect?

Thanks a lot for your help!

**Comment by lijunle** [comment]
This is fixed. Thanks very much! I am closing the bug.

---

In regards to enumerable seq, I don't have real requirement based on that. I understand that, sometime it is hard to change something, sometime it is not worth to implement a feature.

My point is, it is very hard and very wasting time to debug a bug causing page hanging and infinite loop. If it cannot be detected from build time, a runtime exception is fine. It won't cause full page hanging and long-time debugging.

Next guy using enumerable seq gets the exception and google will leads him here. Otherwise, he will spent a full night to debug and find out the cause is the Seq.range.
